aspnetcore/security/samesite.md

```diff
@@ -0,0 +1,145 @@
+
+## SameSite draft standard
+
+The [SameSite 2016 draft](https://tools.ietf.org/html/draft-west-first-party-cookies-07#section-4.1) extension to HTTP cookies:
+
+* Was intended to mitigate cross site request forgery (CSRF).
+* Was designed as a feature servers would opt into by adding the new "SameSite" attribute and attribute values to cookies.
+* Is supported in ASP.NET 2.0 and later.
+
+## New (2019) SameSite draft standard
+
+The new [SameSite 2019 draft](https://tools.ietf.org/html/draft-west-cookie-incrementalism-00):
+
+* Is not backwards compatible.
+* Cookies should be treated as `SameSite=Lax` by default.
+* Cookies that explicitly assert `SameSite=None` in order to enable cross-site delivery should be marked as `Secure`. `None` is a new entry to opt out.
+
+`Lax` is OK for most application cookies but breaks cross site scenarios like OpenIdConnect and Ws-Federation login. Most [OAuth](https://oauth.net/) logins are not affected due to differences in how the request flows. The new `None` parameter causes compatibility problems with clients that implemented the prior draft standard (for example, iOS 12). Chrome browsers (Chrome 80) are expected to go live in February 2020 with the 2019 SameSite draft standard.
+
+::: moniker range="= aspnetcore-3.1"
+ASP.NET Core 3.1 provides the following SameSite support:
+
+* Redefines the behavior of `SameSiteMode.None` to emit `SameSite=None`
+* Adds a new value `SameSiteMode.Unspecified` to omit the `SameSite` attribute.
+* All cookies APIs default to `Unspecified`. Some components that use cookies set values more specific to their scenarios, for example
+  * The `OpenIdConnect` correlation.
+  * `nonce` cookies.
+
+::: moniker-end
+
+::: moniker range=">= aspnetcore-3.0"
+
+Each ASP.NET Core component that emits cookies need to decide if `SameSite` is appropriate. ASP.NET Core 3.0 and later aligns SameSite defaults with the new draft standard. The following API's have changed the default from `SameSiteMode.Lax ` to `SameSiteMode.None`:
+
+* <xref:Microsoft.AspNetCore.Http.CookieOptions> used with [HttpContext.Response.Cookies.Append](xref:Microsoft.AspNetCore.Http.IResponseCookies.Append*)
+* <xref:Microsoft.AspNetCore.Http.CookieBuilder>  used as a factory for `CookieOptions`
+* [CookiePolicyOptions.MinimumSameSitePolicy](xref:Microsoft.AspNetCore.Builder.CookiePolicyOptions.MinimumSameSitePolicy)
+
+ALL ASP.NET Core components that emit cookies:
+
+* Override the preceding defaults with settings appropriate for their scenarios.
+* The overridden preceding default values have not changed.
+
+| Component | Default |
+| ------------- | ------------- |
+| <xref:System.Web.HttpContext.Session>  | `Lax` |
+| <xref:Microsoft.AspNetCore.Mvc.ViewFeatures.CookieTempDataProvider>   | `Lax` |
+| `Antiforgery` | `Strict` |
```

Comment:
@serpent5  do you know what this and the next 4 components are?

Reply:
1.  [`IAntiforgery`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.antiforgery.iantiforgery) or [`AntiforgeryOptions.Cookie`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.antiforgery.antiforgeryoptions.cookie)
1. [`AddCookie`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.cookieextensions.addcookie) or [`CookieAuthenticationOptions.Cookie`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.cookies.cookieauthenticationoptions.cookie#Microsoft_AspNetCore_Authentication_Cookies_CookieAuthenticationOptions_Cookie)
1. [`AddTwitter`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.twitterextensions.addtwitter) or [`TwitterOptions.StateCookie`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.twitter.twitteroptions.statecookie#Microsoft_AspNetCore_Authentication_Twitter_TwitterOptions_StateCookie)
1. [`RemoteAuthenticationHandler`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.remoteauthenticationhandler-1) or [`RemoteAuthenticationOptions.CorrelationCookie`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.remoteauthenticationoptions.correlationcookie).
1. [`AddOpenIdConnect`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.openidconnectextensions.addopenidconnect) or [`OpenIdConnectOptions.NonceCookie`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.openidconnect.openidconnectoptions.noncecookie#Microsoft_AspNetCore_Authentication_OpenIdConnect_OpenIdConnectOptions_NonceCookie)